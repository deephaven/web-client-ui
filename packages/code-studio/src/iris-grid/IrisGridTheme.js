import IrisGridTheme from './IrisGridTheme.module.scss';

export default Object.freeze({
  backgroundColor: IrisGridTheme['grid-bg'],
  font: IrisGridTheme.font,
  headerBackgroundColor: IrisGridTheme['header-bg'],
  headerColor: IrisGridTheme['header-color'],
  headerSeparatorColor: IrisGridTheme['header-separator-color'],
  headerSeparatorHoverColor: IrisGridTheme['header-separator-hover-color'],
  headerHiddenSeparatorHoverColor:
    IrisGridTheme['header-hidden-separator-hover-color'],
  headerSortBarColor: IrisGridTheme['header-sort-bar-color'],
  headerReverseBarColor: IrisGridTheme['header-reverse-bar-color'],
  headerBarCasingColor: IrisGridTheme['header-bar-casing-color'],
  headerFont: IrisGridTheme['header-font'],
  rowBackgroundColors: IrisGridTheme['row-background-colors'],
  rowHoverBackgroundColor: IrisGridTheme['row-hover-bg'],
  selectionColor: IrisGridTheme['selection-color'],
  selectionOutlineColor: IrisGridTheme['selection-outline-color'],
  selectionOutlineCasingColor: IrisGridTheme['selection-outline-casing-color'],
  selectedRowHoverBackgroundColor: IrisGridTheme['selected-row-hover-bg'],
  scrollBarBackgroundColor: IrisGridTheme['scroll-bar-bg'],
  scrollBarHoverBackgroundColor: IrisGridTheme['scroll-bar-hover-bg'],
  scrollBarCasingColor: IrisGridTheme['scroll-bar-casing-color'],
  scrollBarCornerColor: IrisGridTheme['scroll-bar-corner-color'],
  scrollBarColor: IrisGridTheme['scroll-bar-color'],
  scrollBarHoverColor: IrisGridTheme['scroll-bar-hover-color'],
  scrollBarActiveColor: IrisGridTheme['scroll-bar-active-color'],
  textColor: IrisGridTheme['text-color'],
  positiveNumberColor: IrisGridTheme['positive-number-color'],
  negativeNumberColor: IrisGridTheme['negative-number-color'],
  zeroNumberColor: IrisGridTheme['zero-number-color'],
  dateColor: IrisGridTheme['date-color'],
  filterBarActiveBackgroundColor: IrisGridTheme['filter-bar-active-bg'],
  filterBarExpandedBackgroundColor: IrisGridTheme['filter-bar-expanded-bg'],
  filterBarExpandedActiveBackgroundColor:
    IrisGridTheme['filter-bar-expanded-active-bg'],
  filterBarExpandedActiveCellBackgroundColor:
    IrisGridTheme['filter-bar-expanded-active-cell-bg'],
  filterBarSeparatorColor: IrisGridTheme['filter-bar-separator-color'],
  filterBarActiveColor: IrisGridTheme['filter-bar-active-color'],
  filterBarErrorColor: IrisGridTheme['filter-bar-error-color'],
  filterIconColor: IrisGridTheme['filter-icon-color'],
  scrimColor: IrisGridTheme['scrim-color'],
  contextMenuSortIconColor: IrisGridTheme['context-menu-sort-icon-color'],
  contextMenuReverseIconColor: IrisGridTheme['context-menu-reverse-icon-color'],

  allowRowResize: false,
  autoSelectRow: true,
  gridColumnColor: null,
  gridRowColor: null,
  groupedColumnDividerColor: IrisGridTheme['grouped-column-divider-color'],
  columnHoverBackgroundColor: null,
  headerHorizontalPadding: 12,
  scrollBarSize: 11,
  scrollBarHoverSize: 15, // system default scrollbar width is 17
  minScrollHandleSize: 24,
  rowHeight: 19,
  columnWidth: 100,
  rowHeaderWidth: 0,
  rowFooterWidth: 60,
  columnHeaderHeight: 30,
  filterBarHeight: 30, // includes 1px casing at bottom
  filterBarCollapsedHeight: 5, // includes 1px casing at bottom
  sortHeaderBarHeight: 2,
  reverseHeaderBarHeight: 4,
  filterBarHorizontalPadding: 4,

  // Amount of blur to apply to the bottom of the scrim while animating in
  scrimBlurSize: 25,
  minScrimTransitionTime: 150,
  maxScrimTransitionTime: 350,

  scrollSnapToRow: true,

  linkerColumnHoverBackgroundColor: IrisGridTheme['linker-column-hover-bg'],

  treeLineColor: IrisGridTheme['tree-line-color'],
  treeMarkerColor: IrisGridTheme['tree-marker-color'],
  treeMarkerHoverColor: IrisGridTheme['tree-marker-hover-color'],

  floatingGridColumnColor: null,
  floatingGridRowColor: IrisGridTheme['floating-grid-row-color'],
  floatingRowBackgroundColors: IrisGridTheme['floating-row-background-colors'],
  floatingDividerInnerColor: IrisGridTheme['floating-divider-inner-color'],
  floatingDividerOuterColor: IrisGridTheme['floating-divider-outer-color'],
});
