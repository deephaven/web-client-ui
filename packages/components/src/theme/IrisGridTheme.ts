// TODO NEEDS TYPE DEFINITIONS WITHOUT CREATING A CYCLIC DEPENDENCY

const IrisGridTheme = Object.freeze({
  backgroundColor: '--spectrum-gray-50',
  white: '--spectrum-global-color-static-white',
  black: '--spectrum-global-color-static-black',
  font: '12px Fira Sans, Helvetica, Arial, sans-serif',

  headerBackgroundColor: '--spectrum-gray-100',
  headerColor: '--spectrum-alias-heading-text-color',
  headerSeparatorColor: '--spectrum-gray-50',
  headerSeparatorHoverColor: '--spectrum-gray-600',
  headerHiddenSeparatorHoverColor: '--spectrum-accent-color-700',
  headerSortBarColor: '--spectrum-purple-visual-color',
  headerReverseBarColor: '--spectrum-green-visual-color',
  headerBarCasingColor: '--spectrum-gray-50',
  headerFont: '600 12px Fira Sans, Helvetica, Arial, sans-serif',

  rowBackgroundColors: '--spectrum-gray-200 --spectrum-gray-100',
  rowHoverBackgroundColor: '--spectrum-alias-highlight-hover',
  selectionColor: '--spectrum-alias-highlight-selected',
  selectionOutlineColor: '--spectrum-accent-color-900',
  selectionOutlineCasingColor: '--spectrum-gray-50',
  selectedRowHoverBackgroundColor: '--spectrum-alias-highlight-selected-hover',

  scrollBarBackgroundColor: '--spectrum-gray-100',
  scrollBarHoverBackgroundColor: '--spectrum-gray-200',
  scrollBarCasingColor: '--spectrum-gray-400',
  scrollBarCornerColor: '--spectrum-gray-100',
  scrollBarColor: '--spectrum-gray-500',
  scrollBarHoverColor: '--spectrum-gray-600',
  scrollBarActiveColor: '--spectrum-gray-700',
  scrollBarSelectionTickColor: '--spectrum-alias-highlight-selected-hover',
  scrollBarActiveSelectionTickColor: '--spectrum-accent-color-400',

  textColor: '--spectrum-alias-text-color',
  hyperlinkColor: '--spectrum-accent-color-1000',
  positiveNumberColor: '--spectrum-positive-visual-color',
  negativeNumberColor: '--spectrum-negative-visual-color',
  zeroNumberColor: '--spectrum-notice-visual-color',
  dateColor: '--spectrum-notice-visual-color',
  pendingTextColor: '--spectrum-yellow-1300',
  errorTextColor: '--spectrum-negative-visual-color',
  nullStringColor: '--spectrum-alias-text-color-disabled',

  filterBarActiveBackgroundColor: '--spectum-accent-color-300',
  filterBarExpandedBackgroundColor: '--spectrum-gray-200',
  filterBarExpandedActiveBackgroundColor: '--spectum-accent-color-100',
  filterBarExpandedActiveCellBackgroundColor: '--spectum-accent-color-200',
  filterBarSeparatorColor: '--spectrum-gray-600',
  filterBarActiveColor: '--spectrum-accent-color-600',
  filterBarErrorColor: '--spectrum-negative-visual-color',
  filterIconColor: '--spectrum-accent-color-600',

  scrimColor: '--spectrum-gray-50',
  contextMenuSortIconColor: '--spectrum-purple-visual-color',
  contextMenuReverseIconColor: '--spectrum-green-visual-color',

  allowRowResize: false,
  autoSelectRow: true,
  gridColumnColor: null,
  gridRowColor: null,
  groupedColumnDividerColor: '--spectrum-gray-50',
  columnHoverBackgroundColor: null,
  headerHorizontalPadding: 12,
  scrollBarSize: 13,
  scrollBarHoverSize: 16, // system default scrollbar width is 17
  minScrollHandleSize: 24,
  rowHeight: 19,
  columnWidth: 100,
  rowHeaderWidth: 0,
  rowFooterWidth: 60,
  columnHeaderHeight: 30,
  filterBarHeight: 30, // includes 1px casing at bottom
  filterBarCollapsedHeight: 5, // includes 1px casing at bottom
  sortHeaderBarHeight: 2,
  reverseHeaderBarHeight: 4,
  filterBarHorizontalPadding: 4,

  // Amount of blur to apply to the bottom of the scrim while animating in
  scrimBlurSize: 25,
  minScrimTransitionTime: 150,
  maxScrimTransitionTime: 350,

  scrollSnapToRow: true,

  linkerColumnHoverBackgroundColor: '--spectrum-alias-highlight-selected-hover',

  treeLineColor: '--spectrum-gray-700',
  treeMarkerColor: '--spectrum-gray-800',
  treeMarkerHoverColor: '--spectrum-gray-900',

  floatingGridColumnColor: null,
  floatingGridRowColor: '--spectrum-gray-300',
  floatingRowBackgroundColors: '--spectrum-gray-200',
  floatingDividerInnerColor: '--spectrum-gray-200',
  floatingDividerOuterColor: '--spectrum-gray-50',

  overflowButtonColor: '--spectrum-gray-700',
  overflowButtonHoverColor: '--spectrum-gray-900',

  zeroLineColor: '--spectrum-gray-300',
  positiveBarColor: '--spectrum-positive-visual-color',
  negativeBarColor: '--spectrum-negative-visual-color',
});

export default IrisGridTheme;
