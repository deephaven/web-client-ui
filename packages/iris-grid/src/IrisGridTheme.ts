import {
  getExpressionRanges,
  resolveCssVariablesInRecord,
} from '@deephaven/components';
import type { GridThemeType } from '@deephaven/grid';
import { GridTheme } from '@deephaven/grid';
import Log from '@deephaven/log';
import { GridColor, NullableGridColor } from '@deephaven/grid/src/GridTheme';
import { ColorUtils } from '@deephaven/utils';
import IrisGridThemeRaw from './IrisGridTheme.module.scss';

const log = Log.module('IrisGridTheme');

export type IrisGridThemeType = GridThemeType & {
  filterBarCollapsedHeight: number;
  filterBarHeight: number;
  reverseHeaderBarHeight: number;
  filterIconColor: string;
  filterBarActiveColor: GridColor;
  contextMenuSortIconColor: GridColor;
  contextMenuReverseIconColor: GridColor;
  minScrimTransitionTime: number;
  maxScrimTransitionTime: number;
  nullStringColor: GridColor;
  pendingTextColor: GridColor;
  dateColor: GridColor;
  positiveNumberColor: GridColor;
  negativeNumberColor: GridColor;
  zeroNumberColor: GridColor;
  errorTextColor: GridColor;
  groupedColumnDividerColor: GridColor;
  linkerColumnHoverBackgroundColor: GridColor;
  scrimBlurSize: number;
  scrimColor: GridColor;
  headerReverseBarColor: GridColor;
  sortHeaderBarHeight: number;
  headerSortBarColor: GridColor;
  headerBarCasingColor: GridColor;
  filterBarExpandedActiveBackgroundColor: GridColor;
  filterBarExpandedBackgroundColor: GridColor;
  filterBarSeparatorColor: GridColor;
  filterBarExpandedActiveCellBackgroundColor: GridColor;
  filterBarErrorColor: GridColor;
  filterBarHorizontalPadding: number;
  filterBarActiveBackgroundColor: GridColor;
  overflowButtonColor: GridColor;
  overflowButtonHoverColor: GridColor;
  floatingGridRowColor: NullableGridColor;
};

/**
 * Derive default Iris grid theme from IrisGridThemeRaw. Note that CSS variables
 * contained in IrisGridThemeRaw are resolved to their actual values. This means
 * that the returned theme is statically defined and does not change when CSS
 * variables change.
 */
export function createDefaultIrisGridTheme(): IrisGridThemeType {
  const IrisGridTheme = resolveCssVariablesInRecord(IrisGridThemeRaw);

  // row-background-colors is a space-separated list of colors, so we need to
  // normalize each color expression in the list individually
  IrisGridTheme['row-background-colors'] = getExpressionRanges(
    IrisGridTheme['row-background-colors'] ?? ''
  )
    .map(([start, end]) =>
      ColorUtils.normalizeCssColor(
        IrisGridTheme['row-background-colors'].substring(start, end + 1)
      )
    )
    .join(' ');

  log.debug2('Iris grid theme:', IrisGridThemeRaw);
  log.debug2('Iris grid theme derived:', IrisGridTheme);

  return Object.freeze({
    ...GridTheme,
    backgroundColor: IrisGridTheme['grid-bg'],
    white: IrisGridTheme.white,
    black: IrisGridTheme.black,
    font: IrisGridTheme.font,
    headerBackgroundColor: IrisGridTheme['header-bg'],
    headerColor: IrisGridTheme['header-color'],
    headerSeparatorColor: IrisGridTheme['header-separator-color'],
    headerSeparatorHoverColor: IrisGridTheme['header-separator-hover-color'],
    headerHiddenSeparatorHoverColor:
      IrisGridTheme['header-hidden-separator-hover-color'],
    headerSortBarColor: IrisGridTheme['header-sort-bar-color'],
    headerReverseBarColor: IrisGridTheme['header-reverse-bar-color'],
    headerBarCasingColor: IrisGridTheme['header-bar-casing-color'],
    headerFont: IrisGridTheme['header-font'],
    rowBackgroundColors: IrisGridTheme['row-background-colors'],
    rowHoverBackgroundColor: IrisGridTheme['row-hover-bg'],
    selectionColor: IrisGridTheme['selection-color'],
    selectionOutlineColor: IrisGridTheme['selection-outline-color'],
    selectionOutlineCasingColor:
      IrisGridTheme['selection-outline-casing-color'],
    selectedRowHoverBackgroundColor: IrisGridTheme['selected-row-hover-bg'],
    scrollBarBackgroundColor: IrisGridTheme['scroll-bar-bg'],
    scrollBarHoverBackgroundColor: IrisGridTheme['scroll-bar-hover-bg'],
    scrollBarCasingColor: IrisGridTheme['scroll-bar-casing-color'],
    scrollBarCornerColor: IrisGridTheme['scroll-bar-corner-color'],
    scrollBarColor: IrisGridTheme['scroll-bar-color'],
    scrollBarHoverColor: IrisGridTheme['scroll-bar-hover-color'],
    scrollBarActiveColor: IrisGridTheme['scroll-bar-active-color'],
    scrollBarSelectionTickColor: IrisGridTheme['selected-row-hover-bg'],
    scrollBarActiveSelectionTickColor:
      IrisGridTheme['scroll-bar-active-selection-tick-color'],
    textColor: IrisGridTheme['text-color'],
    hyperlinkColor: IrisGridTheme['hyperlink-color'],
    positiveNumberColor: IrisGridTheme['positive-number-color'],
    negativeNumberColor: IrisGridTheme['negative-number-color'],
    zeroNumberColor: IrisGridTheme['zero-number-color'],
    dateColor: IrisGridTheme['date-color'],
    pendingTextColor: IrisGridTheme['pending-text-color'],
    errorTextColor: IrisGridTheme['error-text-color'],
    nullStringColor: IrisGridTheme['null-string-color'],
    filterBarActiveBackgroundColor: IrisGridTheme['filter-bar-active-bg'],
    filterBarExpandedBackgroundColor: IrisGridTheme['filter-bar-expanded-bg'],
    filterBarExpandedActiveBackgroundColor:
      IrisGridTheme['filter-bar-expanded-active-bg'],
    filterBarExpandedActiveCellBackgroundColor:
      IrisGridTheme['filter-bar-expanded-active-cell-bg'],
    filterBarSeparatorColor: IrisGridTheme['filter-bar-separator-color'],
    filterBarActiveColor: IrisGridTheme['filter-bar-active-color'],
    filterBarErrorColor: IrisGridTheme['filter-bar-error-color'],
    filterIconColor: IrisGridTheme['filter-icon-color'],
    scrimColor: IrisGridTheme['scrim-color'],
    contextMenuSortIconColor: IrisGridTheme['context-menu-sort-icon-color'],
    contextMenuReverseIconColor:
      IrisGridTheme['context-menu-reverse-icon-color'],

    allowRowResize: false,
    autoSelectRow: true,
    gridColumnColor: null,
    gridRowColor: null,
    groupedColumnDividerColor: IrisGridTheme['grouped-column-divider-color'],
    columnHoverBackgroundColor: null,
    headerHorizontalPadding: 12,
    scrollBarSize: 13,
    scrollBarHoverSize: 16, // system default scrollbar width is 17
    minScrollHandleSize: 24,
    rowHeight: parseInt(IrisGridTheme['row-height'], 10) || 19, // IrisGrid test breaks without the fallback value
    columnWidth: 100,
    rowHeaderWidth: 0,
    rowFooterWidth: 60,
    columnHeaderHeight: parseInt(IrisGridTheme['header-height'], 10) || 30,
    filterBarHeight: 30, // includes 1px casing at bottom
    filterBarCollapsedHeight: 5, // includes 1px casing at bottom
    sortHeaderBarHeight: 2,
    reverseHeaderBarHeight: 4,
    filterBarHorizontalPadding: 4,

    shadowAlpha: parseFloat(IrisGridTheme['row-shadow-alpha']) || 0.15,

    // Amount of blur to apply to the bottom of the scrim while animating in
    scrimBlurSize: 25,
    minScrimTransitionTime: 150,
    maxScrimTransitionTime: 350,

    scrollSnapToRow: true,

    linkerColumnHoverBackgroundColor: IrisGridTheme['linker-column-hover-bg'],

    treeLineColor: IrisGridTheme['tree-line-color'],
    treeMarkerColor: IrisGridTheme['tree-marker-color'],
    treeMarkerHoverColor: IrisGridTheme['tree-marker-hover-color'],

    floatingGridColumnColor: null,
    floatingGridRowColor: IrisGridTheme['floating-grid-row-color'],
    floatingRowBackgroundColors:
      IrisGridTheme['floating-row-background-colors'],
    floatingDividerInnerColor: IrisGridTheme['floating-divider-inner-color'],
    floatingDividerOuterColor: IrisGridTheme['floating-divider-outer-color'],

    overflowButtonColor: IrisGridTheme['overflow-button-color'],
    overflowButtonHoverColor: IrisGridTheme['overflow-button-hover-color'],

    zeroLineColor: IrisGridTheme['zero-line-color'],
    positiveBarColor: IrisGridTheme['positive-bar-color'],
    negativeBarColor: IrisGridTheme['negative-bar-color'],
    markerBarColor: IrisGridTheme['marker-bar-color'],
  });
}
