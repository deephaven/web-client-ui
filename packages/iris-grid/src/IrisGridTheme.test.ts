import { createDefaultIrisGridTheme } from './IrisGridTheme';

jest.mock('@deephaven/components', () => ({
  ...jest.requireActual('@deephaven/components'),
  resolveCssVariablesInRecord: jest.fn(
    () =>
      new Proxy(
        {},
        {
          get(_target, name) {
            return `IrisGridTheme['${String(name)}']`;
          },
        }
      )
  ),
}));

beforeEach(() => {
  jest.clearAllMocks();
  expect.hasAssertions();
});

describe('createDefaultIrisGridTheme', () => {
  it('should derive the default Iris grid theme', () => {
    const theme = createDefaultIrisGridTheme();
    expect(theme).toMatchSnapshot();
  });
});
