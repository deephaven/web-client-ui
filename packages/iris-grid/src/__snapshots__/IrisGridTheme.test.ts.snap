// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDefaultIrisGridTheme should derive the default Iris grid theme 1`] = `
{
  "activeCellSelectionBorderWidth": 2,
  "allowColumnReorder": true,
  "allowColumnResize": true,
  "allowRowReorder": true,
  "allowRowResize": false,
  "autoSelectColumn": false,
  "autoSelectRow": true,
  "autoSizeColumns": true,
  "autoSizeRows": true,
  "backgroundColor": "IrisGridTheme['grid-bg']",
  "black": "IrisGridTheme['black']",
  "cellHorizontalPadding": 5,
  "columnHeaderHeight": 30,
  "columnHoverBackgroundColor": null,
  "columnWidth": 100,
  "contextMenuReverseIconColor": "IrisGridTheme['context-menu-reverse-icon-color']",
  "contextMenuSortIconColor": "IrisGridTheme['context-menu-sort-icon-color']",
  "dataBarHorizontalPadding": 90,
  "dateColor": "IrisGridTheme['date-color']",
  "density": {
    "compact": {
      "cellHorizontalPadding": 5,
      "columnHeaderHeight": 26,
      "font": "11px Fira Sans, sans-serif",
      "headerHorizontalPadding": 10,
      "iconSize": 14,
      "minColumnWidth": 10,
      "rowHeight": 16,
    },
    "spacious": {
      "cellHorizontalPadding": 7,
      "headerHorizontalPadding": 15,
      "rowHeight": 28,
    },
  },
  "errorTextColor": "IrisGridTheme['error-text-color']",
  "filterBarActiveBackgroundColor": "IrisGridTheme['filter-bar-active-bg']",
  "filterBarActiveColor": "IrisGridTheme['filter-bar-active-color']",
  "filterBarCollapsedHeight": 5,
  "filterBarErrorColor": "IrisGridTheme['filter-bar-error-color']",
  "filterBarExpandedActiveBackgroundColor": "IrisGridTheme['filter-bar-expanded-active-bg']",
  "filterBarExpandedActiveCellBackgroundColor": "IrisGridTheme['filter-bar-expanded-active-cell-bg']",
  "filterBarExpandedBackgroundColor": "IrisGridTheme['filter-bar-expanded-bg']",
  "filterBarFont": "IrisGridTheme['font']",
  "filterBarHeight": 30,
  "filterBarHorizontalPadding": 4,
  "filterBarSeparatorColor": "IrisGridTheme['filter-bar-separator-color']",
  "filterIconColor": "IrisGridTheme['filter-icon-color']",
  "floatingDividerInnerColor": "IrisGridTheme['floating-divider-inner-color']",
  "floatingDividerOuterColor": "IrisGridTheme['floating-divider-outer-color']",
  "floatingGridColumnColor": null,
  "floatingGridRowColor": "IrisGridTheme['floating-grid-row-color']",
  "floatingRowBackgroundColors": "IrisGridTheme['floating-row-background-colors']",
  "font": "IrisGridTheme['font']",
  "gridColumnColor": null,
  "gridRowColor": null,
  "groupedColumnDividerColor": "IrisGridTheme['grouped-column-divider-color']",
  "headerBackgroundColor": "IrisGridTheme['header-bg']",
  "headerBarCasingColor": "IrisGridTheme['header-bar-casing-color']",
  "headerColor": "IrisGridTheme['header-color']",
  "headerFont": "IrisGridTheme['header-font']",
  "headerHiddenSeparatorHoverColor": "IrisGridTheme['header-hidden-separator-hover-color']",
  "headerHiddenSeparatorSize": 5,
  "headerHorizontalPadding": 12,
  "headerResizeHiddenSnapThreshold": 8,
  "headerResizeSnapThreshold": 10,
  "headerReverseBarColor": "IrisGridTheme['header-reverse-bar-color']",
  "headerSeparatorColor": "IrisGridTheme['header-separator-color']",
  "headerSeparatorHandleSize": 5,
  "headerSeparatorHoverColor": "IrisGridTheme['header-separator-hover-color']",
  "headerSortBarColor": "IrisGridTheme['header-sort-bar-color']",
  "hyperlinkColor": "IrisGridTheme['hyperlink-color']",
  "iconSize": 16,
  "linkerColumnHoverBackgroundColor": "IrisGridTheme['linker-column-hover-bg']",
  "markerBarColor": "IrisGridTheme['marker-bar-color']",
  "maxColumnWidth": 600,
  "maxDepth": 6,
  "maxScrimTransitionTime": 350,
  "minColumnWidth": 55,
  "minRowHeight": 20,
  "minScrimTransitionTime": 150,
  "minScrollHandleSize": 24,
  "negativeBarColor": "IrisGridTheme['negative-bar-color']",
  "negativeNumberColor": "IrisGridTheme['negative-number-color']",
  "nullStringColor": "IrisGridTheme['null-string-color']",
  "overflowButtonColor": "IrisGridTheme['overflow-button-color']",
  "overflowButtonHoverColor": "IrisGridTheme['overflow-button-hover-color']",
  "pendingTextColor": "IrisGridTheme['pending-text-color']",
  "positiveBarColor": "IrisGridTheme['positive-bar-color']",
  "positiveNumberColor": "IrisGridTheme['positive-number-color']",
  "reorderOffset": 2,
  "reverseHeaderBarHeight": 4,
  "rowBackgroundColors": "IrisGridTheme['row-background-colors']",
  "rowFooterWidth": 60,
  "rowHeaderWidth": 0,
  "rowHeight": 19,
  "rowHoverBackgroundColor": "IrisGridTheme['row-hover-bg']",
  "scrimBlurSize": 25,
  "scrimColor": "IrisGridTheme['scrim-color']",
  "scrollBarActiveColor": "IrisGridTheme['scroll-bar-active-color']",
  "scrollBarActiveSelectionTickColor": "IrisGridTheme['scroll-bar-active-selection-tick-color']",
  "scrollBarBackgroundColor": "IrisGridTheme['scroll-bar-bg']",
  "scrollBarCasingColor": "IrisGridTheme['scroll-bar-casing-color']",
  "scrollBarCasingWidth": 1,
  "scrollBarColor": "IrisGridTheme['scroll-bar-color']",
  "scrollBarCornerColor": "IrisGridTheme['scroll-bar-corner-color']",
  "scrollBarHoverBackgroundColor": "IrisGridTheme['scroll-bar-hover-bg']",
  "scrollBarHoverColor": "IrisGridTheme['scroll-bar-hover-color']",
  "scrollBarHoverSize": 16,
  "scrollBarSelectionTick": true,
  "scrollBarSelectionTickColor": "IrisGridTheme['selected-row-hover-bg']",
  "scrollBarSize": 13,
  "scrollSnapToColumn": false,
  "scrollSnapToRow": true,
  "selectedColumnHoverBackgroundColor": "#494949",
  "selectedRowHoverBackgroundColor": "IrisGridTheme['selected-row-hover-bg']",
  "selectionColor": "IrisGridTheme['selection-color']",
  "selectionOutlineCasingColor": "IrisGridTheme['selection-outline-casing-color']",
  "selectionOutlineColor": "IrisGridTheme['selection-outline-color']",
  "shadowAlpha": 0.15,
  "shadowBlur": 8,
  "shadowColor": "#000000",
  "sortHeaderBarHeight": 2,
  "textColor": "IrisGridTheme['text-color']",
  "treeDepthIndent": 10,
  "treeHorizontalPadding": 5,
  "treeLineColor": "IrisGridTheme['tree-line-color']",
  "treeMarkerColor": "IrisGridTheme['tree-marker-color']",
  "treeMarkerHoverColor": "IrisGridTheme['tree-marker-hover-color']",
  "white": "IrisGridTheme['white']",
  "zeroLineColor": "IrisGridTheme['zero-line-color']",
  "zeroNumberColor": "IrisGridTheme['zero-number-color']",
}
`;
